File: cart/src/main/scala/de/leanovate/dose/cart/model/CartItems.scala
package de.leanovate.dose.cart.model

case class CartItems(
                      items: Seq[CartItem],
                      valid: Boolean,
                      totalCents: Int
                      )

object CartItems {
  def apply(items: Seq[CartItem]): CartItems = {
    CartItems(
      items = items,
      valid = items.forall(item => item.product.isDefined && item.priceInCent.isDefined),
      totalCents = items.foldLeft(0) {
        (total, item) => total + item.priceInCent.getOrElse(0)
      }
    )
  }
}

File: billing/src/main/java/de/leanovate/dose/billing/model/CartItems.java
package de.leanovate.dose.billing.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class CartItems {

    @JsonProperty
    public List<CartItem> items;

    @JsonProperty
    public boolean valid;

    @JsonProperty
    public int totalCents;

    @Override
    public String toString() {

        return "CartItems{" +
                "items=" + items +
                ", valid=" + valid +
                ", totalCents=" + totalCents +
                '}';
    }
}
